src/internal/m365/backup.go

```diff
@@ -34,6 +35,7 @@ import (
 func (ctrl *Controller) ProduceBackupCollections(
 	ctx context.Context,
 	bpc inject.BackupProducerConfig,
+	rp kinject.RestoreProducer,
```

Comment:
Hmm, I have significant concerns about this pattern.  At large, it produces three smells: 1/ it couples the requirements of one service onto the design of _all_ providers and services; 2/ it requires an interface whose behavior is the _opposite_ of the current functions behavior, which is all kinds of confusing at a glance (it's only incidental that retrieving metadata is a "restore" behavior; meanwhile this func declaration now reads like, "this backup will restore data"); 3/ it goes against the established pattern of metadata collection production and passdown (ie: generically, at the top of the operation layer) by shifting _some_of the that code deep into backup creation.

I think the smells can be fixed up with a fairly minimal amount of work.  Mostly re-thinking how you approach the creation and handoff of this metadata.  High level, right now you're saying, "I'll get the metadata by handing the collection restorer into the backup".  Instead, how about, "I'll get the metadata by having the provider/service export an interface-compliant func for extra metadata retrieval"?

Something like this: 
* add a new func to `inject.BackupProducer` like `RetrieveMetadata(rp RestoreProducer, stuff...)`.  you'll probably want it to return a map[string]whatever to namespace by subservice or path, like the site ID.
* have m365/controller comply with this interface.
* in the controller func, switch on the service.  If it's groups, call into `collectSiteMetadata()` and any other necessary calls.  All other services return nil.
* call `RetrieveMetadata()` in `operations.produceManifestsAndMetadata()`
* extend the produceBackup with a new parameter, "additionalMetadata".  Or maybe make a struct to hold the two metadatas.  Pass the metadata produced in operations down into the backup production.

This way, you're solving the smells by: 1/ keeping apis more consistent with structures that all services use, 2/ ensuring the parameters passed into backup production make sense to backup behavior, 3/ consolidating ownership of the place where behavior is utilized (not necessarily where implementation is held, the behavior utilization is more important here) in a centralized spot.